### DISCUSSION TITLE
Maintain the efficiency and purity of 'dotnet interactive'

### DISCUSSION THREAD

**Discussion by nczsl** [dotnet/interactive#3764] [General]
I suggest not giving users the suggestion to install the "polyglot" plugin; When I learned that I could work with C # on Jupyter in VSCode without installing it, I didn't hesitate to uninstall it. The 'polyglot' plugin has too many completely redundant features for me; I just hope that the functionality of Jupyter C # will not be hindered by it, and that it can maintain focus on quality and update speed in C #, without letting other things affect the release of Jupyter C #. I think updates are very important, so I suggest cutting the two apart, which may maintain the purity, flexibility, and better quality of "dotnet interactive"; Another issue is fixing some bugs now, as the jupyter running the button multiple times on a certain cell may cause the "dotnet interactive" kernel to crash and require a kernel restart; Also, regarding the issue of global "net host" in the background when starting ipynb files in Jupyter, this is my guess; the main purpose is to avoid multiple ipynb files starting multiple "net hosts" in the background at the same time, which would consume too much computer resources

**Comment by jonsequitur** [comment]
Polyglot Notebooks and .NET Interactive are two separate components. Polyglot Notebooks depends on .NET Interactive but not the other way around. .NET Interactive (i.e. the `dotnet-interactive` tool) is a standalone tool and can be used as a Jupyter kernel with other Jupyter frontends. However, the functionality in Polyglot Notebooks supports a number of things that are not supported by Jupyter, e.g. the ability to have choose different languages/subkernels from one cell to another, a variable explorer that's aware of multiple kernels, the support for a JavaScript kernel, etc.

> The 'polyglot' plugin has too many completely redundant features for me; I just hope that the functionality of Jupyter C # will not be hindered by it, and that it can maintain focus on quality and update speed in C #, without letting other things affect the release of Jupyter C #. 

Can you elaborate? And what are you referring to by "Jupyter C#"? Do you mean using the .NET Interactive kernel directly the VS Code Jupyter extension?

>  Another issue is fixing some bugs now, as the jupyter running the button multiple times on a certain cell may cause the "dotnet interactive" kernel to crash and require a kernel restart; 

If you have a stable way to repro this, could you open an issue with instructions?

> Also, regarding the issue of global "net host" in the background when starting ipynb files in Jupyter, this is my guess; the main purpose is to avoid multiple ipynb files starting multiple "net hosts" in the background at the same time, which would consume too much computer resources

These are different kernel instances. Resource usage will be roughly the same if the host is Jupyter.
